src/Workspaces/Core/Portable/CodeRefactorings/ExportCodeRefactoringProviderAttribute.cs

```diff
@@ -15,7 +18,8 @@ public sealed class ExportCodeRefactoringProviderAttribute : ExportAttribute
         /// <summary>
         /// The name of the <see cref="CodeRefactoringProvider"/>.  
         /// </summary>
-        public string Name { get; set; }
+        [DisallowNull]
```

Comment:
This feels like this needs either a doc comment or just a regular comment. Is this the "we could have allowed null and can't change it, but we discourage it?"

Reply:
> Is this the "we could have allowed null and can't change it, but we discourage it?"

It means the property latches non-null (cannot be set back to null once it's set to non-null).